### ISSUE TITLE
[feature request] Send audio files as voice message

### ISSUE THREAD

**Issue by beppe9000** [telegramdesktop/tdesktop#2483]
### Expected behaviour

A file selection window appears and the user selects an audio file (mp3,wav,ogg,etc.). The file is transcoded to the proper codec and sent as a voice message.
### Actual behaviour

There is no way to select a file and send it as voice message

**Comment by hemee** [comment]
Actually vkmusicbot implements something like this to send music. When you request a track client says it's sending a voice message.

**Comment by diazbastian** [comment]
It sounds pretty good. 

In a first stage it could be that all OPUS file is sent as a voice message (as when sending a WebP image as sticker).

**Comment by beppe9000** [comment]
> It sounds pretty good.
> 
> In a first stage it could be that all OPUS file is sent as a voice message (as when sending a WebP image as sticker).

Well, that already happens: I was tired of converting every file to wav and then to opus, so I opened this issue. We need the second stage! ;)

**Comment by diazbastian** [comment]
@beppe9000 I see, I miss the update when this started happening...
I think the features "send as video" or "send as a voice message" are very necessary today.

**Comment by beppe9000** [comment]
Why not just use a prefix like `voice_`? Afaik the file name is lost in the process of sending a voice message.
For example `voice_123.ogg` is sent transcoded (if necessary) and sent as voice message, while anything else is sent normally.

**Comment by tdesktop-closer** [comment]
Hey there!

We're automatically closing this issue since there was no activity in this issue since 401 days ago. We therefore assume that the user has lost interest or resolved the problem on their own. Closed issues that remain inactive for a long period may get automatically locked.

Don't worry though; if this is in error, let us know with a comment and we'll be happy to reopen the issue.

Thanks!

<sub>(Please note that this is an [automated](https://github.com/auchri/github_auto_closer) comment.)</sub>

**Comment by beppe9000** [comment]
@tdesktop-closer Please re-open, this is a feature suggestion so it should be kept so until actually implemented.

**Comment by beppe9000** [comment]
@tdesktop-closer Please re-open, this is a feature suggestion so it should be kept so until actually implemented by someone more involved with the project.

**Comment by beppe9000** [comment]
@tdesktop-closer Please re-open, this is a feature suggestion so it should be kept so until actually implemented by someone more involved with the project.

**Comment by futpib** [comment]
I don't think this is a good idea. I prefer to be sure that a voice message has actually been recorded through a mic (even though it's not 100% guaranteed right now, I've never seen (or have I just not noticed?) anyone fake a voice message (except myself via puleaudio)).

**Comment by stek29** [comment]
@futpib I disagree -- there's already a way to easily "fake" voice messages since TG has open API (and even easier with botapi)

**Comment by stek29** [comment]
oops, didn't mean to close it

**Comment by Aokromes** [comment]
wanted

**Comment by beppe9000** [comment]
one of the worst bots ever created

**Comment by Aokromes** [comment]
> one of the worst bots ever created

no it's not, it helps to find already fixed issues.

**Comment by beppe9000** [comment]
i hate this bot because people configure it to just close old issues

**Comment by Hypfer** [comment]
@beppe9000 
> boast big "issue closed" metrics.

Speaking as a FOSS maintainer, I must say that your perception is wrong here.
Having a huge backlog of issues can be very frustrating and paralyzing for the project and all developers involved, because you just don't know where to start. It's simply too much so you start ignoring the issue tracker at all because it's basically useless.

Therefore, instead of having hundreds of issues which won't be worked on anyways, people employ this bot as a strategy to filter out the noise.
Personaly, I too am not a fan of this bot because it's a bad experience for everyone including users that actually have opened valid issues and now have to work against this bot for no good reason.

That being said, I also don't know how else one would tackle this problem without adding a shitload of manpower to the project, which is often impossible. Especially with foss hobby projects.

In any case, I'm fairly certain that the bot hasn't been employed here just to have some kind of metric since no one is doing business here. It's just a desperate attempt to clean up the huge 800+ issue backlog.
The alternative would be to not do that and forever keep the issue tracker in this broken state which would be even more harmful.

If you're unhappy about this, consider supporting the project and working through the open issues, checking if they were ever  valid, checking if they're still valid etc.

**Comment by Aokromes** [comment]
> Personaly, I too am not a fan of this bot because it's a bad experience for everyone including users that actually have opened valid issues and now have to work against this bot for no good reason.


well, like i said on another tickets, it helps us to find already fixed issues still opened.

**Comment by beppe9000** [comment]
> Personaly, I too am not a fan of this bot because it's a bad experience for everyone including users that actually have opened valid issues and now have to work against this bot for no good reason.

I feel you. It's like having a to-do list that deletes things after 1 month you forget to do them... so stressful. 🤦🏻

@Aokromes i wonder if there are many or little false positives tho

**Comment by Aokromes** [comment]
> little false positives

there are few, but they can be reopened.

**Comment by beppe9000** [comment]
i guess we'll live with that, then

**Comment by proninyaroslav** [comment]
f I sent an ogg file, then on the Android client it appears as a voice message, but on the desktop it still looks like an audio file. I think it's necessary to bring this behavior to the behavior of the mobile client.
![2023-09-27 17-49-23](https://github.com/telegramdesktop/tdesktop/assets/7840559/d5428e4a-07e4-452c-a9a6-7a890b339779)
![2023-09-27_17-50-17](https://github.com/telegramdesktop/tdesktop/assets/7840559/a724ae89-a5b5-442d-82dc-5fd8bfb4b8fb)

**Comment by AntonSamokat** [comment]
Just in case for the bot: still still wanted

**Comment by Aokromes** [comment]
> Just in case for the bot: still still wanted

bot doesn't slates enhancements.

**Comment by AntonSamokat** [comment]
> bot doesn't slates enhancements.

Right, it does not plan enhancements. But as history of this ticket shows, it tries to close the ticket in case of some time of inactivity. Just for that case I simulated slight activity for it to keep the ticket open until it is done.

This seems to be "client management in some way" (facepalm) - to watch their tickets not to be closed after passing 1 month or whatever period.

I do not know may be such strategy makes dev team happy because of regular backlog zeroing but such noise messages from bot and from clients who regularly says "wanted" or creating tickets dups (because of auto closed by bot the previous ones due to timeouts) is not good either.

**Comment by Aokromes** [comment]
> > bot doesn't slates enhancements.
> 
> Right, it does not plan enhancements. But as history of this ticket shows, it tries to close the ticket in case of some time of inactivity. Just for that case I simulated slight activity for it to keep the ticket open until it is done.
> 
> This seems to be "ticket management by client in some way" (facepalm) — to watch their tickets not to be closed after passing 1 month or whatever period.
> 
> I do not know may be such strategy makes dev team happy because of regular backlog zeroing but such noise messages from bot and from clients who regularly says "wanted" or creating tickets dups (because of auto closed by bot the previous ones due to timeouts) is not good either.

that was before a change, now no longer does.

**Comment by AntonSamokat** [comment]
> that was before a change, now no longer does.
> 
> https://i.imgur.com/P8Mf06O.png 2 years without bot activity, and another more before your comment.

I see. That is great.
